### PR TITLE
Gui: support dragging of toolbar inside status/menu bar

### PR DESCRIPTION
**Pull request by realthunder**
Continuation of PR #13571. @PaddleStroke

### CODE REVIEW

File: src/Gui/ToolBarManager.cpp
Comment (kadet1090) [inline src/Gui/ToolBarManager.cpp:315]: While I do understand benefit of having our own ToolBar widget (it would come handy for a lot of reasons) - I don't understand why you create this class now while its only method is a simple call to parent implementation. I'd suggest removing this method and adding other useful helpers like `float()`. Override the moveable behavior etc.

File: src/Gui/ToolBarManager.cpp
Comment (kadet1090) [inline src/Gui/ToolBarManager.cpp:1225]: Creating new object without assigning it to anything is a very strange construction. Probably it is fine as it has properly defined parent so Qt will take care of it but nonetheless it is quite odd - please at least put comment over it that it is intentional and how it is supposed to work.

File: src/Gui/ToolBarManager.cpp
Comment (kadet1090) [inline src/Gui/ToolBarManager.cpp:392]: While you don't have to mark signals with explicit `Q_EMIT` it is nice at it clearly states the intention of that call. Also a comment why it is outside of the block above (to not block this signal - a bug that you have in the previous PR) would be nice as it once again is hard to quickly understand why every call is in that block and it is outside.

**Review by kadet1090 — CHANGES_REQUESTED** [review]

**Comment by PaddleStroke** [comment]
@realthunder thanks ! I have not tested yet, but from the title I guess it is enabling drag from the statusbar and menubar ? I'll test soon.
Btw have you seen the bug report by Syres in the previous PR? Here are the link of the forum topic : https://forum.freecad.org/viewtopic.php?p=758767#p758767

Is this issue fixed in this PR?

**Comment by PaddleStroke** [comment]
I have tested the functionality and it works great. Much better than the previous 'unlock toolbar' menu.

One small note : 
Before with the 'Corner widget' option, the tabbar was added 'Raw' as the corner widget. Resulting in no margin on top/bottom.
Since #13571, the tabbar is now inside a toolbar, and it's the toolbar that is put as corner widget, resulting in some margins : 
image

I don't know if it is possible, but if toolbars that are in the menubar/statusbar could have no top/bottom margins it would look better.

**Comment by PaddleStroke** [comment]
Also I think that you may have introduced https://github.com/FreeCAD/FreeCAD/issues/13927 when fixing the sketcher toolbar problem in https://github.com/FreeCAD/FreeCAD/pull/13571

I have not bisected so I'm not 100% sure but I have a build from before and one after your PR, and it the bug is only in the build after.

I would guess that a preference is not initialized correctly?

**Comment by FEA-eng** [comment]
@realthunder Can you have a look at the proposed changes? Also, there are some conflicts that have to be resolved.

**Comment by kadet1090** [comment]
Not in that state. This is another PR that will require a lot of refactoring after merging it.

**Comment by kadet1090** [comment]
> Hence the requested changes. Do you have more?

Yes and no, depends on how you look at it. I did comment on the stuff that is most important and in my opinion should be changed or clarified before merging so in that way no. But I have at least twice as much questions to the PR that are less important and I decided to simply not flood this PR with comments.

**Comment by FEA-eng** [comment]
Ok, we have to wait for Realthunder’s feedback anyway.

**Comment by PaddleStroke** [comment]
Alternatively, if you already knows what needs to be done (as it appears in the review you made) perhaps we can merge it and you can PR the necessary fixes @kadet1090 ?

**Comment by FEA-eng** [comment]
@PaddleStroke As long as the PR is open and non-draft, it should be considered for merging even after the freeze date.

**Comment by realthunder** [comment]
I am a bit busy these few days. Feel free to change the code as you like.

**Comment by kadet1090** [comment]
> Alternatively, if you already knows what needs to be done (as it appears in the review you made) perhaps we can merge it and you can PR the necessary fixes @kadet1090 ?

I'd rather create another PR to replace this one with fixes included to not risk that it will stay like this and fight with potential conflicts that can happen due to others fixing the same piece of code (which happened with #13571). That said, I also won't have time to do that until next saturday - after that I should have more time for FreeCAD. 

Let's keep this open for now, like it was said - it will get merged as this is marked as ready for review so it can be merged even after the feature freeze.

**Comment by FEA-eng** [comment]
> Let's keep this open for now, like it was said - it will get merged as this is marked as ready for review so it can be merged even after the feature freeze.

Yes. but it can't be merged ATM - there are conflicts that must be resolved. Theoretically, if a new PR (even based on this one) is opened after the freeze, it won't be considered. As long as it counts as UI change only (then the deadline is 24.06) or as an exception.

**Pull request closed without merging**
